lib/sign_in/idme/service.rb

```diff
@@ -48,14 +48,20 @@ def user_info(token)
 
       private
 
-      def get_public_jwks
-        unless config.public_jwks
-          response = perform(:get, config.public_jwks_path, nil, nil)
-          config.public_jwks = parse_public_jwks(response:)
-          Rails.logger.info('[SignIn][Idme][Service] Get Public JWKs Success')
-        end
+      def public_jwks
+        @public_jwks ||= begin
+          cache_hit = true
+
+          response = Rails.cache.fetch(config.jwks_cache_key, expires_in: config.jwks_cache_expiration) do
+            cache_hit = false
+            perform(:get, config.public_jwks_path, nil, nil)
+          end
 
-        config.public_jwks
+          parsed_public_jwks = parse_public_jwks(response:)
+          Rails.logger.info("[SignIn][Idme][Service] Get Public JWKs Success - #{cache_hit ? 'Cache' : 'Request'}")
```

Comment:
Is there a reason not to cache the `parsed_public_jwks`?

Also wouldn't it be simpler to just move this log (or make a similar log) inside the cache block and then you don't need the `cache_hit` variable?

Reply:
Yeah I guess we could just cache `parsed_public_jwks` since we don't need anything else from the response.

On the log, I'm not sure if that would work unless we don't care about keeping track of what was an external request vs cache. If it is cached the block doesn't get executed. So if we had logs inside and outside the block, both would get logged on a cache miss. Did you have something else in mind?

Reply:
Basically I only care that it logs when the request is made. I don't think it's that important to log when the cache hit properly and no request was made.

Reply:
sounds good